tests/network/vmi_istio.go

```diff
@@ -0,0 +1,207 @@
+		vmiDefaultPortsAppCaseName: {
+			appSelector: vmiDefaultPortsAppSelector,
+			ports:       []v1.Port{},
+		},
+		vmiExplicitPortsAppCaseName: {
+			appSelector: vmiExplicitPortsAppSelector,
+			ports:       explicitPorts,
+		},
+	}
+
+	Describe("Virtual machine in a service mesh using masquerade interface with explicit port list", func() {
+		tests.BeforeAll(func() {
+			if !istioServiceMeshDeployed() {
+				Skip("Istio service mesh is required for service-mesh tests to run")
+			}
+
+			tests.BeforeTestCleanup()
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			tests.AddLabelToNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel, "enabled")
+
+			for appCaseKey, vmiCase := range vmiAppCases {
+				vmi := newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(
+					vmiCase.appSelector,
+					network,
+					vmiCase.ports,
+				)
+				svc := newServiceWithAppSelector(
+					fmt.Sprintf("%s-service", vmi.Name),
+					vmi.Namespace,
+					vmiCase.appSelector,
+				)
+				vmiAppCases[appCaseKey].vmi = vmi
+				vmiAppCases[appCaseKey].service = svc
+
+				_, err = virtClient.CoreV1().Services(svc.Namespace).Create(context.Background(), svc, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+				_, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
+				Expect(err).ToNot(HaveOccurred())
+			}
+			for _, vmiCase := range vmiAppCases {
+				tests.WaitUntilVMIReady(vmiCase.vmi, libnet.WithIPv6(console.LoginToCirros))
+				for _, targetPort := range []int{svcDeclaredTestPort, svcUndeclaredTestPort} {
+					tests.StartHTTPServer(vmiCase.vmi, targetPort)
+				}
+			}
+		})
+		table.DescribeTable("should be able to reach", func(destination string, port int) {
+			var srcVMI, dstVMI *v1.VirtualMachineInstance
+			switch destination {
+			case vmiExplicitPortsAppCaseName:
+				srcVMI = vmiAppCases[vmiDefaultPortsAppCaseName].vmi
+				dstVMI = vmiAppCases[vmiExplicitPortsAppCaseName].vmi
+			case vmiDefaultPortsAppCaseName:
+				srcVMI = vmiAppCases[vmiExplicitPortsAppCaseName].vmi
+				dstVMI = vmiAppCases[vmiDefaultPortsAppCaseName].vmi
+			}
+			dstVMI, err := virtClient.VirtualMachineInstance(dstVMI.Namespace).Get(dstVMI.Name, &metav1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			dstVMIAddress := dstVMI.Status.Interfaces[0].IP
+
+			By(fmt.Sprintf("checking that service running on port %d can be reached", port))
+			err = console.RunCommand(srcVMI, fmt.Sprintf("curl %s:%d", dstVMIAddress, port), 10*time.Second)
+			Expect(err).ShouldNot(HaveOccurred())
+		},
+			table.Entry("declared port in VMI with explicit ports", vmiExplicitPortsAppCaseName, svcDeclaredTestPort),
+			table.Entry("undeclared port in  VMI with explicit ports", vmiExplicitPortsAppCaseName, svcUndeclaredTestPort),
+			table.Entry("declared port in VMI with default masquerade interface", vmiDefaultPortsAppCaseName, svcDeclaredTestPort),
```

Comment:
I already commented about it but the comment is somewhere outdated.
I'm not sure how the last two scenarios are working. We don't have a code passing data from/to localhost to a vm with no specific ports (you call it - default masquerade interface), right? So how does this work?

Reply:
As we discussed, added a [condition](https://github.com/kubevirt/kubevirt/compare/e47924b601c29ede8f8b175d6ede514fe9f480bb..2a20abaa3692d559d53ec3a034cd7465209461f0) to adding prerouting dnat rule only if istio annotation is not present.

e2e tests now exercise the following:
- inbound traffic can reach VM
- inbound traffic from outside the mesh cannot reach VM if Istio PeerAuthenticaton policy is set to strict, which verifies that inbound traffic is indeed being proxied
- outbound traffic can reach http://kubevirt.io
- outbound traffic can not reach http://kubevirt.io if Istio OutboundTrafficPolicy is set to REGISTRY_ONLY, which verifies that the outbound traffic is indeed being proxied